src/PlaneWaveBasis.jl

```diff
@@ -121,10 +121,29 @@ build_kpoints(basis::PlaneWaveBasis, kcoords) =
 # This is the "internal" constructor; the higher-level one below should be preferred
 @timing function PlaneWaveBasis(model::Model{T}, Ecut::Number,
                                 kcoords::AbstractVector, ksymops, symops=nothing;
-                                fft_size=determine_grid_size(model, Ecut), variational=true) where {T <: Real}
-    @assert Ecut > 0
+                                fft_size=nothing, variational=true, optimize_fft_size=false, supersampling=2) where {T <: Real}
+    if variational
+        @assert Ecut > 0
+        if fft_size === nothing
+            fft_size = determine_fft_size(model, Ecut; supersampling=supersampling)
+        end
+    else
+        # ensure fft_size is provided, and other options are not set
+        @assert fft_size !== nothing
+        @assert supersampling == 2
+        @assert !optimize_fft_size
```

Comment:
I know you like assertions, but since this is very user-facing I think a slightly more verbose error message is appropriate in my opinion.

Reply:
This is not very user-facing: the user should not mess with these unless they know what they're doing. I can add some doc for these options though

Reply:
This is a pretty internal thing for tests, it's not a recommended thing (yet). Once we understand more about these methods we can see

Reply:
Then I would not documented it, but make a TODO next to the assertions that this should be converted into error messages once this has stabilised.